cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -183,6 +186,49 @@ public void pushApp(FullHttpRequest request, HttpResponder responder,
     }
   }
 
+  /**
+   * Pushes application configs of requested applications to linked repository in Json format.
+   * It expects a post body that has a list of application ids and an optional commit message
+   * E.g.
+   *
+   * <pre>
+   * {@code
+   * {
+   *   "appIds": ["app_id_1", "app_id_2"],
+   *   "commitMessage": "pushed application XYZ"
+   * }
+   * }
+   *
+   * </pre>
+   * The response will be a {@link OperationMeta} object, which encapsulates the application name,
+   * version and fileHash.
+   */
+  @POST
+  @Path("/apps/push")
+  public void pushApps(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespaceId) throws Exception {
+    checkSourceControlMultiFeatureFlag();
+    NamespaceId namespace = validateNamespaceId(namespaceId);
+
+    PushMultipleAppsRequest appsRequest;
+    try {
+      appsRequest = parseBody(request, PushMultipleAppsRequest.class);
+    } catch (JsonSyntaxException e) {
+      throw new BadRequestException("Invalid request body.", e);
```

Comment:
In this particular case we want to include e.getMessage() in the error message, since that is what gets displayed to the user. 

Note that this is not always what we want, especially for non-user facing exceptions, as that leads to duplicate logging.